### PR TITLE
Batched fixes for march 2022 release 

### PR DESCRIPTION
**Pull request by vvolkl**
BEGINRELEASENOTES
- Batched fixes for march 2022 release 
- Add run environment to build environment to allow running tests for framework packages
- Some more explicit env var settings as CPATH, LIBRARY_PATH have been removed from the run environment in upstream spack
- Round of new tags

ENDRELEASENOTES

### CODE REVIEW

File: packages/fccanalyses/package.py [hunk 1]
```diff
@@ -24,8 +26,15 @@ class Fccanalyses(CMakePackage, Key4hepPackage):
     depends_on('edm4hep')
     depends_on('py-awkward')
     depends_on('fcc-edm', when="@:0.2.9")
-    depends_on('acts@5.00.0', when="@0.3.0:")
+    depends_on('acts', when="@0.3.0:")
```
[the remark this answers is not in the document]
Comment (vvolkl) [reply packages/fccanalyses/package.py:29]: Yes, that should probably be a bit more specific, but pinning version 6.00.0 is not necessary. Can you change that to 
```
depends_on('acts@6.00.0:', when='@0.3.5:')
```

[the remark this answers is not in the document]
Comment (tmadlener) [reply packages/fccanalyses/package.py:29]: Good catch. I was actually aiming for the open range ;)

File: packages/fccanalyses/package.py [hunk 2]
```diff
@@ -24,8 +26,16 @@ class Fccanalyses(CMakePackage, Key4hepPackage):
     depends_on('edm4hep')
     depends_on('py-awkward')
```
Comment (tmadlener) [inline packages/fccanalyses/package.py:27]:
```suggestion
    depends_on('py-awkward@1.4.0')
```
There were some changes to the c++ interface after that, that would need fixes in FCCAnalyses

File: packages/k4simdelphes/package.py [hunk 3]
```diff
@@ -30,15 +32,17 @@ class K4simdelphes(CMakePackage, Ilcsoftpackage):
     variant('delphes_hepmc', default=True, description='Build standalone executable with Hepmc input.')
     variant('delphes_pythia_evtgen', default=True, description='Build standalone executable with Pythia+EvtGen input')
 
-    depends_on('edm4hep')
-    depends_on('delphes@3.4.3pre10:')
+    depends_on('edm4hep', type=('build', 'link', 'run'))
+    depends_on('podio', type=('build', 'link', 'run'))
+    depends_on('delphes@3.4.3pre10:', type=('build', 'link', 'run'))
```
Comment (tmadlener) [inline packages/k4simdelphes/package.py:37]:
```suggestion
    depends_on('delphes@3.5:', type=('build', 'link', 'run'))
```
Should I try to make a more fine-grained thing about which k4simdelphes version needs which delphes version? Or are we satisfied with the fact that probably only the latest will build in any case?

Comment (vvolkl) [reply packages/k4simdelphes/package.py:37]: Given that there are several delphes versions which work but may give wrong results, I think it'll be good to be explicit here. But there's also no need to keep all of the pre-releases in the recipes - I'll take a look at it in a future PR.

**Pull request merged**